src/item_factory.cpp

```diff
@@ -3410,7 +3410,7 @@ void islot_comestible::deserialize( const JsonObject &jo )
     optional( jo, was_loaded, "calories", default_nutrition.calories );
 
     optional( jo, was_loaded, "contamination", contamination,
-              weighted_string_id_reader<diseasetype_id, float> {1.0} );
+              weighted_string_id_reader<diseasetype_id, float> {static_cast<float>( 1.0 )} );
```

Comment:
There's a less verbose language feature for this.
```suggestion
              weighted_string_id_reader<diseasetype_id, float> {1.0f} );
```

Reply:
Much better!